cdk/kraken/src/cdk.ts

```diff
@@ -81,7 +81,7 @@ export class CDKPublishStack extends Stack {
         yarn package
         mv dist/js/*.tgz dist/js/kraken.tgz
         yarn publish --non-interactive --access public dist/js/kraken.tgz`,
-        if: `github.ref == 'refs/heads/${fullConfig.defaultBranch}'`,
+        if: `github.ref == 'refs/heads/${fullConfig.defaultBranch}' && "$(npm info @pennlabs/kittyhawk version)" == "$(node -p "require('./package.json').version")"`,
```

Comment:
I was just trying to solve this myself. I think we should do something more similar to `yarn info @pennlabs/kittyhawk versions --json | jq -e '.data | index("0.0.0")'`. Which will error out if 0.0.0 is not found in the list of published versions.

The reason I like this a bit more is because A. it's using yarn and B. it checks all published versions on npm rather than just the latest version.

Reply:
I also don't know if we can run commands in this `if` statement, so we may have to do it as a first step in the job itself

Reply:
Ohh I like your `yarn info` solution too. We can also use `yarn version --non-interactive --json` which just returns 

```{"type":"info","data":"Current version: 0.7.1"}```